test/SerialIOTest.cpp

```diff
@@ -7522,6 +7522,8 @@ void groupbased_read_write(std::string const &ext)
 
         E_x.setAttribute("updated_in_run", 0);
         E_y.setAttribute("updated_in_run", 0);
+        E_y.setAttribute("changed_datatype_in_run", 0);
+        write.close();
```

Comment:
`.close()` calls added for stricter error check, the destructors will swallow exceptions.